### PR TITLE
Add an example to create a custom converter for a NMF transformer

### PR DESCRIPTION
**Pull request by xadupre**

### CODE REVIEW

File: docs/api_summary.rst [hunk 1]
```diff
@@ -12,8 +12,35 @@ in *scikit-onnx*.
 Converters
 ==========
 
+Both functions convert a *scikit-learn* model into ONNX.
+The first one lets the user manually 
+define the inputs name and types. The second one
```
[the remark this answers is not in the document]
Comment (xadupre) [reply docs/api_summary.rst]: Fixed.

File: docs/api_summary.rst [hunk 2]
```diff
@@ -12,8 +12,35 @@ in *scikit-onnx*.
...
+infers this information from the training data.
+These two functions are the main entry points to converter.
+The rest of the API is needed if a model has not converter
```
[the remark this answers is not in the document]
Comment (xadupre) [reply docs/api_summary.rst]: Fixed.

File: docs/api_summary.rst [hunk 3]
```diff
@@ -12,8 +12,35 @@ in *scikit-onnx*.
...
+or created from scratch.
+
 .. autofunction:: skl2onnx.convert_sklearn
 
+.. autofunction:: skl2onnx.to_onnx
+
+Register a new converter
+========================
+
+If a model has not converter
```
[the remark this answers is not in the document]
Comment (xadupre) [reply docs/api_summary.rst]: Fixed.

File: docs/examples/plot_nmf.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+Custom Operator for NMF Decomposition
+=====================================
+
+`NMF <https://scikit-learn.org/stable/modules/generated/
+sklearn.decomposition.NMF.html>`_ factorizes an input matrix
+into two matrices *W, H* of rank *k* so that :math:`WH \\sim M``.
+:math:`M=(m_{ij})` may be a binary matrix where *i* is a user
+and *j* a product he bought. The prediction
+function depends on whether or not the user needs an
+recommandation for an existing user or a new user.
```
[the remark this answers is not in the document]
Comment (xadupre) [reply docs/examples/plot_nmf.py]: Fixed.

File: skl2onnx/algebra/type_helper.py [hunk 5]
```diff
@@ -39,6 +42,8 @@ def _guess_type(given_type):
     if isinstance(given_type, np.ndarray):
         if given_type.dtype == np.float32:
             return FloatTensorType(given_type.shape)
+        elif given_type.dtype == np.int32:
```
[the remark this answers is not in the document]
Comment (xadupre) [reply skl2onnx/algebra/type_helper.py]: Fixed.

File: docs/examples/plot_nmf.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+"""
+
+import skl2onnx
+import onnxruntime
+import sklearn
+from sklearn.decomposition import NMF
+import numpy as np
+import matplotlib.pyplot as plt
+import os
```
[the remark this answers is not in the document]
Comment (xadupre) [reply docs/examples/plot_nmf.py]: Fixed.

File: docs/examples/plot_nmf.py [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+`NMF <https://scikit-learn.org/stable/modules/generated/
+sklearn.decomposition.NMF.html>`_ factorizes an input matrix
+into two matrices *W, H* of rank *k* so that :math:`WH \\sim M``.
+:math:`M=(m_{ij})` may be a binary matrix where *i* is a user
+and *j* a product he bought. The prediction
+function depends on whether or not the user needs an
+recommandation for an existing user or a new user.
+This example addresses the first case.
+
+The second case is more complex as it theoritically
```
[the remark this answers is not in the document]
Comment (xadupre) [reply docs/examples/plot_nmf.py]: Fixed

File: docs/examples/plot_nmf.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (prabhat00155) [reply docs/examples/plot_nmf.py]: "recommandation" typo still there.

File: docs/examples/plot_nmf.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (prabhat00155) [reply docs/examples/plot_nmf.py]: I don't see the change: `import os` should be at the very top.

**Pull request merged**